### PR TITLE
Introduce standard node affinity support. Replace NFD label scheduling

### PR DESCRIPTION
**Pull request by pacevedom**
Remove NFD label set when using affinity annotations in recipes. Removes all knowledge of NFD labels from the operator, using only standard resources to determine where to schedule workloads that need a specific kernel.
Also addresses problems with skew when upgrading nodes if using NFD.

### CODE REVIEW

**Comment by yevgeny-shnaidman** [comment]
@pacevedom  what is the reason for this PR? Not sure i understand why we need to set the names affinity

**Comment by pacevedom** [comment]
> @pacevedom what is the reason for this PR? Not sure i understand why we need to set the names affinity

This is the second part of removing NFD dependency, also useful for 2.0 as it will have a similar/same mechanism. Using NFD labels might be convenient in some situations but gets you into additional problems like dependencies, configuration, skew when ugprading, etc.
This PR solves all that by using standard k8s ways of scheduling. Now NFD labels are entirely unknown to SRO, only relevant to recipes, if they have such needs.

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -81,12 +80,19 @@ func (ci *clusterInfo) nodeVersionInfo(nodeList *corev1.NodeList) (map[string]No
...
-			OSMajor:        "rhel" + osMajor,
-			OSMajorMinor:   "rhel" + osVersion,
+		data, ok := info[kernelFullVersion]
+		if !ok {
+			data = NodeVersion{
+				OSVersion:      osVersion,
+				ClusterVersion: clusterVersion,
+				OSMajor:        "rhel" + osMajor,
+				OSMajorMinor:   "rhel" + osVersion,
+				Names:          []string{node.Name},
```
Comment (yevgeny-shnaidman) [inline pkg/upgrade/upgrade.go:90]: why do we need the names of the nodes? we are still keeping the kernel full version, and we are getting it from the node, and not from the NFD labels. So why can't we just keep using it for affinity?

Comment (pacevedom) [reply pkg/upgrade/upgrade.go:90]: Using `nodeAffinity` requires either labels or supported fields, and none of these are able to match against `status.kernelVersion` in the nodes. Therefore we need something that we can map to expressions/fields, hence the names.

File: internal/resourcehelper/helper.go [hunk 2]
```diff
@@ -110,36 +110,74 @@ func (rh *resourceHelper) UpdateResourceVersion(req *unstructured.Unstructured,
 
 func (rh *resourceHelper) SetNodeSelectorTerms(obj *unstructured.Unstructured, terms map[string]string) error {
 	switch obj.GetKind() {
-	case "DaemonSet", "Deployment", "Statefulset": // TODO(qbarrand) should this be StatefulSet?:
-		if err := rh.nodeSelectorTerms(terms, obj, "spec", "template", "spec", "nodeSelector"); err != nil {
+	case "DaemonSet", "Deployment", "StatefulSet":
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply internal/resourcehelper/helper.go:113]: Its weird indeed. There is a `nodeSelector` at CRD level that is applied to all resources in the helm chart. Then you can have `nodeSelector` per resource if you cofigure them like that. The CRD option is maybe too coarse to do such a thing, as it applies to `BuildConfig`, for example, which is not really needed.

**Comment by pacevedom** [comment]
/hold
Need to test something with upgrades.

**Comment by ybettan** [comment]
/retest
@pacevedom can you please squash the commits?

**Comment by pacevedom** [comment]
PTAL @qbarrand @yevgeny-shnaidman @ybettan @pmtk @enriquebelarte

**Comment by ybettan** [comment]
Can you please update the commit message to be consistent with the PR description?

**Comment by pacevedom** [comment]
Wait until v2 comes in.

**Comment by bthurber** [comment]
Recommend moving this to v2.

**Pull request closed without merging**